### PR TITLE
Add descriptions about GitHub Actions in the "Useful Developer Tools" page

### PR DESCRIPTION
**Pull request by maropu**
This PR adds descriptions about how to run tests in a forked repository using GitHub Actions.
This comes from https://github.com/apache/spark/pull/29504.

Screen Shot 2020-08-22 at 23 03 19

### CODE REVIEW

**Comment by maropu** [comment]
cc: @HyukjinKwon @dongjoon-hyun

**Comment by HyukjinKwon** [comment]
+1 nice

File: developer-tools.md [hunk 1]
```diff
@@ -228,6 +228,25 @@ Getting logs from the pods and containers directly is an exercise left to the re
...
+this can burden our limited resources of GitHub Actions.
+
+Our script enables you to run tests for a branch in your forked repository.
+Let's say that you have a branch named "your_branch" for a Pull Request.
+To run tests on "your_branch" and check test reuslts:
+
+- Clicks a "Actions" tab in your forked repository.
+- Selects a "Build and test" workfolow in a "All workflows" list.
+- Pushs a "Run workflow" button and enters "your_branch" in a "Target branch to run" field.
+- When a "Build and test" workflow finished, clicks a "Report test results" workfolow to check test results.
```
Comment (kiszk) [inline developer-tools.md]: nit: `workfolow` -> `workflow`?

File: developer-tools.md [hunk 2]
```diff
@@ -228,6 +228,25 @@ Getting logs from the pods and containers directly is an exercise left to the re
 
 Kubernetes, and more importantly, minikube have rapid release cycles, and point releases have been found to be buggy and/or break older and existing functionality.  If you are having trouble getting tests to pass on Jenkins, but locally things work, don't hesitate to file a Jira issue.
 
+<h3>Running tests in your forked repository using GitHub Actions</h3>
+
+GitHub Actions is a functionality within GitHub that enables continuous integration and a wide range of automation.
+We have already started uing some action scripts and one of them is to run tests for Pull Requests.
```
Comment (kiszk) [inline developer-tools.md]: nit: `uing` -> `using`

File: developer-tools.md [hunk 3]
```diff
@@ -228,6 +228,25 @@ Getting logs from the pods and containers directly is an exercise left to the re
...
+If you are planning to create a new Pull Reques, it is important to check if tests can pass on your branch before creating it.
```
Comment (kiszk) [inline developer-tools.md]: nit: `Reques` -> `Request`

Comment (kiszk) [reply developer-tools.md]: suggestion: `before creating it` -> `before creating the Pull Request`

File: developer-tools.md [hunk 4]
```diff
@@ -228,6 +228,25 @@ Getting logs from the pods and containers directly is an exercise left to the re
...
+This is because our GitHub Acrions script automatically run tests for your Pull Request/following commits and
```
Comment (kiszk) [inline developer-tools.md]: nit: `run` -> `runs`?

File: developer-tools.md [hunk 5]
```diff
@@ -228,6 +228,25 @@ Getting logs from the pods and containers directly is an exercise left to the re
...
+this can burden our limited resources of GitHub Actions.
```
Comment (kiszk) [inline developer-tools.md:237]: How about adding rough number of limited resources?

File: developer-tools.md [hunk 6]
```diff
@@ -228,6 +228,25 @@ Getting logs from the pods and containers directly is an exercise left to the re
...
+
+Our script enables you to run tests for a branch in your forked repository.
+Let's say that you have a branch named "your_branch" for a Pull Request.
+To run tests on "your_branch" and check test reuslts:
```
Comment (kiszk) [inline developer-tools.md]: nit: `reuslts` -> `results`

File: site/developer-tools.html [hunk 7]
```diff
@@ -406,6 +406,27 @@ <h4>Testing K8S</h4>
 
 <p>Kubernetes, and more importantly, minikube have rapid release cycles, and point releases have been found to be buggy and/or break older and existing functionality.  If you are having trouble getting tests to pass on Jenkins, but locally things work, don&#8217;t hesitate to file a Jira issue.</p>
 
+<h3>Running tests in your forked repository using GitHub Actions</h3>
+
+<p>GitHub Actions is a functionality within GitHub that enables continuous integration and a wide range of automation.
+We have already started uing some action scripts and one of them is to run tests for Pull Requests.
```
Comment (kiszk) [inline site/developer-tools.html]: ditto at Lines 412-415

File: developer-tools.md (hunk 1 above)
Comment (maropu) [reply developer-tools.md]: Ah, thanks! I forgot to run a typo checker. I run it by myself and pushed it.

File: developer-tools.md (hunk 4 above)
Comment (kiszk) [reply developer-tools.md]: How about this?  Now, it looks that `our ... script ... run ...`

File: developer-tools.md (hunk 5 above)
Comment (maropu) [reply developer-tools.md:237]: I saw the max number, 180, in https://issues.apache.org/jira/browse/SPARK-32264?focusedCommentId=17170040&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17170040 But, this number might change in future, so I think it the current statement is okay for most developers.

File: developer-tools.md (hunk 4 above)
Comment (maropu) [reply developer-tools.md]: Yea, in the following commit, I've updated it.

File: developer-tools.md (hunk 5 above)
Comment (kiszk) [reply developer-tools.md:237]: I see

**Comment by kiszk** [comment]
Looks good to me. Hope someone does another pass.

File: developer-tools.md [hunk 8]
```diff
@@ -228,6 +228,25 @@ Getting logs from the pods and containers directly is an exercise left to the re
 
 Kubernetes, and more importantly, minikube have rapid release cycles, and point releases have been found to be buggy and/or break older and existing functionality.  If you are having trouble getting tests to pass on Jenkins, but locally things work, don't hesitate to file a Jira issue.
 
+<h3>Running tests in your forked repository using GitHub Actions</h3>
+
+GitHub Actions is a functionality within GitHub that enables continuous integration and a wide range of automation.
+We have already started using some action scripts and one of them is to run tests for Pull Requests.
```
Comment (dongjoon-hyun) [inline developer-tools.md]: This looks good. Could you add some link to this from `Pull Request` section in https://spark.apache.org/contributing.html?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (with minor comments).
Thank you so much, @maropu !

File: developer-tools.md (hunk 8 above)
Comment (maropu) [reply developer-tools.md]: Sure!

Comment (maropu) [reply developer-tools.md]: Looks like we don't have the direct link to the secion, so I added the link to the page.

**Comment by maropu** [comment]
Thanks, all! Merged to asf-site.

**Pull request closed without merging**